content/indexes-removal/index.md

```diff
@@ -0,0 +1,182 @@
+---
+rfc: 0024
+start_date: 2018-10-02
+decision_date:
+pr: openregister/registers-rfcs#41
+status: draft
+---
+
+# Indexes removal
+
+## Summary
+
+This RFC proposes the changes to make to the data model and REST API
+consequence of removing the general concept of indexes.
+
+This RFC is a breaking change.
+
+This RFC supersedes RFC0009 (Entry hash) due the changes on the data model
+that affect entries.
+
+For reference,
+[ADR007](https://github.com/openregister/openregister-java/blob/master/doc/arch/adr-007-read-api-changes.md)
+is the document that introduced the indexes changes.
+
+## Motivation
+
+After a period of assessment the Registers team concluded that indexes should
+not be approached the way they are. As a consequence of this decision, we have
+the opportunity to simplify the data model and the REST API, mostly roll back
+to the state it was before the indexes work was introduced.
+
+## Explanation
+
+### Data model
+
+The main change on the data model is the entry. It changes from:
+
+```elm
+type Entry =
+  { number: Integer
+  , timestamp: Timestamp
+  , key: ID
+  , item: Set Hash
```

Comment:
```suggestion
  , blob: Set Hash
```

Reply:
Well, this the example of the entry I'm proposing to change so it has to be "item".

Reply:
Ah, I assumed the starting point for this change would be post-RFC16 (https://github.com/openregister/registers-rfcs/pull/29). Doesn't matter as long as it's consistent.